query-engine/core/src/response_ir/mod.rs

```diff
@@ -133,6 +133,19 @@ impl Item {
         match self {
             Self::Value(pv) => Some(pv),
             Self::Ref(r) => Arc::try_unwrap(r).ok().and_then(|r| r.into_value()),
+            Self::List(list) => {
+                let mut values = vec![];
+
+                for item in list {
+                    if let Some(pv) = item.into_value() {
+                        values.push(pv)
+                    } else {
+                        return None;
+                    }
+                }
+
+                Some(PrismaValue::List(values))
+            }
```

Comment:
This is needed for https://github.com/prisma/prisma-engines/blob/eaaec29a262790019f89b114f2dd2a73cd019772/query-engine/core/src/response_ir/mod.rs#L55, which is used to map the findMany result back to findUnique.

[One of the new tests](https://github.com/prisma/prisma-engines/blob/eaaec29a262790019f89b114f2dd2a73cd019772/query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/batch/select_one_compound.rs#L297-L324) showed that scalar lists were failing because it couldn't convert serialized scalar lists back to prisma values.

This piece of code handles that case.

Reply:
Thanks for the clarification, makes sense!